docs/architecture/accounts.md

```diff
@@ -35,11 +35,11 @@ Storage for user-defined data that is composed of two components.
 
 1. Storage slots: Users have access to `254` slots. Per slot, users can store a `word` (~32 bytes). Currently, we reserve slot `0` for the account owner's key and slot `255` for information about the storage layout.
 
-2. Storage maps: Users requiring additional storage can use the second component a `StorageMap`. A storage map is a key-value store where both keys and values are `word`s (~32 bytes). Thus, a single storage map can store huge amounts of data. Under the hood, storage maps are implemented as Sparse Merkle trees, and the root of the roots of these trees are stored in storage slots. So, a single account can contain up to 254 `StorageMaps`.
+2. Storage maps: Users requiring additional storage can use a `StorageMap`. A storage map is a key-value store where both keys and values are `word`s (~32 bytes). A single storage map can store huge amounts of data, each storage map can store `2^256` `word`s. Under the hood, storage maps are implemented as sparse Merkle trees of depth 64, and the roots of these trees are stored in storage slots. So, a single account can contain up to 254 `StorageMaps`. Note: it is better to use private accounts to store huge amounts of data.
```

Comment:
A few comments:
- I would write the 3rd sentence as: A single storage map can store huge amounts of data (e.g., up to $2^{256}$ `word`s per map).
- I would get rid of the part about depth 64 - I think that's an implementation detail not really relevant for the user.
- For the last sentence, might make sense to make it into a separate paragraph to explain why it is better and who it is better for (otherwise, this sentence raises more questions than answers).